src/Instrumentation/PDO/src/PDOInstrumentation.php

```diff
@@ -329,4 +374,63 @@ private static function isDistributeStatementToLinkedSpansEnabled(): bool
 
         return filter_var(get_cfg_var('otel.instrumentation.pdo.distribute_statement_to_linked_spans'), FILTER_VALIDATE_BOOLEAN, FILTER_NULL_ON_FAILURE) ?? false;
     }
+
+    private static function isSqlCommenterEnabled(): bool
```

Comment:
I think a lot of this code could be moved into its own class, and then be unit tested. It seems like it shouldn't be called "sql commenter" - that was the name of the product before it was donated to otel, but it looks like the objective is to define how to do something similar/better. I guess we're waiting on approved spec changes to see what the names of things should be?

Reply:
Done. Moved to its own class, named `ContextPropagation`. Please see if it looks good. Thanks!

Reply:
If you were to add the sql-server propagation mentioned elsewhere in this PR, would it be in that class, or in a different, similar class? I feel like it should be in a different one, and so maybe `SqlPropagator` is an interface, and `SqlCommentPropagator` is this class, with potentially a `SqlServerPropagator` coming later which uses set_context_info?

Reply:
No going to implement the sql-server propagation in this PR so as to make this scope manageable but I am happy to add the placeholder class `SqlServerPropagator` first.

Done! I added the class `SqlPropagatorInterface`, `SqlCommenterPropagator` & `SqlServerPropagator`.